consensus/safety-rules/src/tests/suite.rs

```diff
@@ -16,114 +17,111 @@ fn make_proposal_with_qc_and_proof(
     round: Round,
     proof: Proof,
     qc: QuorumCert,
-    validator_signer: &ValidatorSigner,
+    signer: &ValidatorSigner,
 ) -> VoteProposal<Round> {
-    test_utils::make_proposal_with_qc_and_proof(round, round, proof, qc, validator_signer)
+    test_utils::make_proposal_with_qc_and_proof(round, round, proof, qc, signer)
 }
 
 fn make_proposal_with_parent(
     round: Round,
     parent: &VoteProposal<Round>,
     committed: Option<&VoteProposal<Round>>,
-    validator_signer: &ValidatorSigner,
+    signer: &ValidatorSigner,
 ) -> VoteProposal<Round> {
-    test_utils::make_proposal_with_parent(round, round, parent, committed, validator_signer)
+    test_utils::make_proposal_with_parent(round, round, parent, committed, signer)
 }
 
-#[test]
-fn test_initial_state() {
+type RoundCallback = fn() -> (Box<dyn TSafetyRules<Round>>, Arc<ValidatorSigner>);
+type ByteArrayCallback = fn() -> (Box<dyn TSafetyRules<Vec<u8>>>, Arc<ValidatorSigner>);
```

Comment:
Why we need to test u64 and Vec<u8> separately? this generic is the Payload of the block and should not matter for safety rules at all?

Reply:
Because I am anal and wanted something a bit more interesting going across the serialization than just a u64...